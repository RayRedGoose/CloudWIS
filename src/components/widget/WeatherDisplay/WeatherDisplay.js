import React from 'react'
import './WeatherDisplay.scss'
import WeatherRow from '../WeatherRow/WeatherRow'

const WeatherDisplay = (props) => {
  return (
    <section className="weather-display">
      <h1 name={props.searchCity}>{props.searchCity}</h1>
      <WeatherRow
        weatherData={props.weatherCurrentData}
        title='Today'
        type='day'
        key='day'/>
      <WeatherRow
        weatherData={props.weatherDaysData}
        title='5 Day Forecast'
        type='fiveDays'
        key='fiveDays'/>
    </section>
  )
}

export default WeatherDisplay
