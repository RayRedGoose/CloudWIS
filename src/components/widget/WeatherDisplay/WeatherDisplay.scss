.weather-display {
  padding: 40px;
  width: 55vw;
  h1 {
    border-bottom: 2px solid #0F1A20;
    color: #fff;
    font-size: 40px;
    text-shadow: 0 0 5px 0 #000;
    position: relative;
    margin-top: 40px;
    padding-bottom: 5px;
    &:after {
      content: attr(name);
      font-size: 4em;
      line-height: 0.8em;
      opacity: 0.1;
      position: absolute;
      bottom: 5px; left: 40px;
    }
  }
}
