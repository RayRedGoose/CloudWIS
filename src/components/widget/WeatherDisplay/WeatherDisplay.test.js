import React from 'react';
import { shallow } from 'enzyme';
import WeatherDisplay from './WeatherDisplay';

it('should match the snapshot with all data passed in correctly', () => {
  const mockProps = {
    weatherCurrentData: {desc: 'current weather info'},
    weatherDaysData: {desc: 'five days weather info'},
    searchCity: 'Denver'
  }

  const block = shallow(
    <WeatherDisplay {...mockProps} />
  );

  expect(block).toMatchSnapshot();
});
