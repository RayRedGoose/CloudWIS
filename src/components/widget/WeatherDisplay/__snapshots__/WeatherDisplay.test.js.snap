// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot with all data passed in correctly 1`] = `
<section
  className="weather-display"
>
  <h1
    name="Denver"
  >
    Denver
  </h1>
  <WeatherRow
    key="day"
    title="Today"
    type="day"
    weatherData={
      Object {
        "desc": "current weather info",
      }
    }
  />
  <WeatherRow
    key="fiveDays"
    title="5 Day Forecast"
    type="fiveDays"
    weatherData={
      Object {
        "desc": "five days weather info",
      }
    }
  />
</section>
`;
